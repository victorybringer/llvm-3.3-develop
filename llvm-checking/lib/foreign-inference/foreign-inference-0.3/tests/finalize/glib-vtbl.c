#include <stdlib.h>

typedef void* gpointer;
typedef size_t gsize;
typedef struct _GMemVTable GMemVTable;

struct _GMemVTable {
  gpointer (*malloc)      (gsize    n_bytes);
  gpointer (*realloc)     (gpointer mem,
      gsize    n_bytes);
  void     (*free)        (gpointer mem);
  /* optional; set to NULL if not used ! */
  gpointer (*calloc)      (gsize    n_blocks,
      gsize    n_block_bytes);
  gpointer (*try_malloc)  (gsize    n_bytes);
  gpointer (*try_realloc) (gpointer mem,
      gsize    n_bytes);
};

static GMemVTable glib_mem_vtable = {
  malloc,
  realloc,
  free,
  calloc,
  malloc,
  realloc,
};

#define G_UNLIKELY(x) x
#define G_LIKELY(x) x

void g_free(gpointer mem) {
  if(G_LIKELY(mem))
    glib_mem_vtable.free(mem);
}
