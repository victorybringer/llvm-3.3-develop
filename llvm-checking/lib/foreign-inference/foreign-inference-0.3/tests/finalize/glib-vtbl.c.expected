fromList [ ("@g_free", fromList ["%mem"]) ]
